Mathlib/Algebra/MonoidAlgebra/MapDomain.lean

```diff
@@ -91,25 +87,18 @@ end MonoidAlgebra
 
 namespace AddMonoidAlgebra
 
-variable {k G}
+section Semiring
+variable [Semiring R] [Semiring S] {f : M → N}
 
-section
-
-variable [Semiring k] [NonUnitalNonAssocSemiring R]
+abbrev mapDomain (f : M → N) (v : R[M]) : R[N] := Finsupp.mapDomain f v
 
-abbrev mapDomain {G' : Type*} (f : G → G') (v : k[G]) : k[G'] :=
-  Finsupp.mapDomain f v
+lemma mapDomain_sum (f : M → N) (s : S[M]) (v : M → S → R[M]) :
+    mapDomain f (s.sum v) = s.sum fun a b ↦ mapDomain f (v a b) := Finsupp.mapDomain_sum
 
-theorem mapDomain_sum {k' G' : Type*} [Semiring k'] {f : G → G'} {s : AddMonoidAlgebra k' G}
-    {v : G → k' → k[G]} :
-    mapDomain f (s.sum v) = s.sum fun a b => mapDomain f (v a b) :=
-  Finsupp.mapDomain_sum
+lemma mapDomain_injective (hf : Injective f) : Injective (mapDomain (R := R) f) :=
+  Finsupp.mapDomain_injective hf
 
-theorem mapDomain_single {G' : Type*} {f : G → G'} {a : G} {b : k} :
-    mapDomain f (single a b) = single (f a) b :=
-  Finsupp.mapDomain_single
```

Comment:
What is your justification for deleting this lemma without deprecation?

Reply:
Bad merge maybe? I agree it should be kept.

Reply:
I was copy-pasting the code from the corresponding `MonoidAlgebra` section, and indeed it turns out we had `AddMonoidAlgebra.mapDomain_single` but not `MonoidAlgebra.mapDomain_single`.